### PR TITLE
Do not generate new models for readonly schema references

### PR DESCRIPTION
**Pull request by mrginglymus**
The changes in #12348 meant that `$ref`s in `allOf` schemas generated new model names. However, a common pattern is to have a referenced schema with the only modification being `readOnly`, which is a change to the parent not the referenced schema. In order to not generate duplicate models, special case this one special case.

A demonstration of the problematic model has been included as a typescript-fetch client.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
can you please update the samples as mentioned in step 3?

https://github.com/OpenAPITools/openapi-generator/runs/8061002912?check_suite_focus=true

**Comment by mrginglymus** [comment]
Ahh, sorry! I've found that the `.openapi-generator-ignore` file seems to be added and removed alternately on eveyr run; I'm not sure if that's a Windows thing or not...

File: samples/client/petstore/typescript-fetch/builds/allOf-readonly/models/Club.ts [hunk 1]
```diff
@@ -28,10 +28,10 @@ import {
 export interface Club {
     /**
      * 
-     * @type {ClubOwner}
+     * @type {Owner}
      * @memberof Club
      */
-    owner?: ClubOwner;
```
Comment (mrginglymus) [inline samples/client/petstore/typescript-fetch/builds/allOf-readonly/models/Club.ts]: Hmm, I hadn't even noticed this bug - it seems like something has dropped the `readOnly` flag in an `allOf` context. That's obviously fixed now here, but may need addressing elsewhere too.

**Comment by wing328** [comment]
tested with java client as well. the result looks ok. will include the fix to see if others have a different opinion/method to fix this.

**Pull request merged**